app-backend/api/src/main/scala/datasource/Routes.scala

```diff
@@ -52,30 +52,34 @@ trait DatasourceRoutes extends Authentication
     get {
       rejectEmptyResponse {
         complete {
-          readOne[Datasource](Datasources.getDatasource(datasourceId))
+          readOne[Datasource](Datasources.getDatasource(datasourceId, user))
         }
       }
     }
   }
 
   def createDatasource: Route = authenticate { user =>
     entity(as[Datasource.Create]) { newDatasource =>
-      onSuccess(write[Datasource](Datasources.insertDatasource(newDatasource, user))) { datasource =>
-        complete(datasource)
+      authorize(user.isInRootOrSameOrganizationAs(newDatasource)) {
```

Comment:
Yes, that is correct. `authenticate` identifies the user, and `authorize` takes a boolean function which allows the request to proceed or not. If `authenticate` fails, the user gets a [`401`](https://httpstatuses.com/401), whereas if `authorize` fails they get a [`403`](https://httpstatuses.com/403)